# Activity 8

This activity tests your skills
in formulating an engineering problem as solving a system of linear algebraic equations,
as well as analyzing results thereof.

# GROUP POWER
- Romel Dave Ayop
- Lee Jire Delos Reyes
- Sum Maquilan

## Scenario

The lateral view of an [inclined strut](./act-08.png)
is designed as a ten-member truss.

You are tasked to analyze a scenario wherein
loads are approximated as concentrated forces 
acting on the non-support joints.
Members $AC$, $CE$, and $EG$ are collinear,
and perpendicular to the lines of action of loads $p_{1}$, $p_{2}$, and $p_{3}$.
Members $BD$, $DF$, and $FG$ are collinear,
and perpendicular
not just to the lines of action of loads $p_{4}$ and $p_{5}$,
but also to the supporting surface.

Let the dimensions, material makeup, and weight of the members be negligible.
Assume that the joints do not resist moments.
Determine the member forces developed
under such a conservative idealization of the strut.

## Modelling

Cast the original task as solving a system of linear algebraic equations
for the forces in $AC$, $BC$, $BD$, $CD$, $CE$, $DE$, $DF$, $EF$, $EG$, and $FG$.

Arrange the equations in the matrix-vector form
$\boldsymbol{K} \boldsymbol{\phi} = \boldsymbol{\delta}$,
where
$\boldsymbol{\phi}$ collects the member forces in the order presented above.
For uniformity, assume that tensile forces are positive.

## Core code

In [1]:
import math as mt

import numpy as np

from math import sin, cos, atan, pi, sqrt

Define a Python function
`get_coeffs_rhs()`
that takes four positional arguments
(namely, `d`, `h`, `theta`, `p1`, `p2`, `p3`, `p4`, `p5`,
in that order)
and returns two NumPy arrays
(namely,
`coeff_matrix`, `rhs_vector`
in that order).
The roles of these inputs and outputs should be evident.

In [2]:
def get_coeffs_rhs(d, h, theta, p1, p2, p3, p4, p5):

    a = atan(h / (3*d))
    b = atan((2*h) / (3*d))
    t = theta * (pi / 180)
    y = (pi/2) - t - a

    q  = (pi/2) - t
    qa = q + a
    y2a = y + 2*a

    K = np.array([
        [-cos(y),      -cos(y+a+b),     0,     cos(t),     cos(y),        0,        0,     0,     0,       0],
        [-sin(y),      -sin(y+a+b),     0,    -sin(t),     sin(y),        0,        0,     0,     0,       0],
        [     0,             0,     -cos(q),  -cos(t),        0,     cos(qa),   cos(q),    0,     0,       0],
        [     0,             0,     -sin(q),   sin(t),        0,     sin(qa),   sin(q),    0,     0,       0],
        [     0,             0,         0,          0,   -cos(y),   -cos(y2a),   0,     cos(t),  cos(y),   0],
        [     0,             0,         0,          0,   -sin(y),   -sin(y2a),   0,    -sin(t),  sin(y),   0],
        [     0,             0,         0,          0,       0,          0,   -cos(q),  -cos(t),   0,   cos(q)],
        [     0,             0,         0,          0,       0,          0,   -sin(q),   sin(t),   0,   sin(q)],
        [     0,             0,         0,          0,       0,          0,      0,       0,   -cos(y), -cos(q)],
        [     0,             0,         0,          0,       0,          0,      0,       0,   -sin(y), -sin(q)]
    ])

    RHS = np.array([
        -p1 * cos(t + a),
         p1 * sin(t + a),

        -p5 * cos(t),
         p5 * sin(t),

        -p2 * cos(t + a),
         p2 * sin(t + a),

        -p4 * cos(t),
         p4 * sin(t),

        -p3 * cos(t + a),
         p3 * sin(t + a)
    ])

    return K, RHS

## Assessment

Define Python variables
`D`, `H`, `THETA`, `P1`, `P2`, `P3`, `P4`, and `P5`
for storing the values of parameters
$d$, $h$, $\theta$, $p_{1}$, $p_{2}$, $p_{3}$, $p_{4}$, and $p_{5}$,
respectively.

Every time you solve for $\boldsymbol{\phi}$,
- use `get_coeffs_rhs()` to generate the coefficient matrix and the right-hand-side vector,
  respectively storing them in Python variables `K` and `d`;
- show the determinant and the condition number of the coefficient matrix;
  and
- print $\boldsymbol{\phi}$.


### Task 1

Determine the member forces when $C$ is loaded by a unit force and all other joints are unloaded.

Define a Python variable `phi_1` to store $\boldsymbol{\phi}$ for this task.

In [3]:
D = 1.0
H = 6.0
THETA = 15.0

P1 = 1.0
P2 = 0.0
P3 = 0.0
P4 = 0.0
P5 = 0.0

K, d = get_coeffs_rhs(D, H, THETA, P1, P2, P3, P4, P5)
detK  = np.linalg.det(K)
condK = np.linalg.cond(K)
phi_1 = np.linalg.solve(K, d)

print("Determinant of K:", detK)
print("Condition number of K:", condK)
print("phi_1 = [", ", ".join(f"{x: .4f}" for x in phi_1), "]")

Determinant of K: 0.46566840006975907
Condition number of K: 6.542591657117212
phi_1 = [ -1.1667, -1.5366, -0.0000, -0.0000, -0.0000, -0.0000, -0.0000, -0.0000, -0.0000, -0.0000 ]


### Task 2

Determine the member forces when $E$ is loaded by a unit force and all other joints are unloaded.

Define a Python variable `phi_2` to store $\boldsymbol{\phi}$ for this task.

In [4]:
D     = 1.0
H     = 6.0
THETA = 15.0

P1 = 0.0
P2 = 1.0
P3 = 0.0
P4 = 0.0
P5 = 0.0

K, dvec = get_coeffs_rhs(D, H, THETA, P1, P2, P3, P4, P5)

detK  = np.linalg.det(K)
condK = np.linalg.cond(K)
phi_2 = np.linalg.solve(K, dvec)

print("Determinant of K:", detK)
print("Condition number of K:", condK)
print("phi_2 = [", ", ".join(f"{x: .4f}" for x in phi_2), "]")

Determinant of K: 0.46566840006975907
Condition number of K: 6.542591657117212
phi_2 = [ -0.3333, -0.7683, -0.5590,  1.1180, -0.7500, -1.2500, -0.0000, -0.0000, -0.0000, -0.0000 ]


### Task 3

Determine the member forces when $G$ is loaded by a unit force and all other joints are unloaded.

Define a Python variable `phi_3` to store $\boldsymbol{\phi}$ for this task.

In [5]:
D     = 1.0
H     = 6.0
THETA = 15.0

P1 = 0.0
P2 = 0.0
P3 = 1.0
P4 = 0.0
P5 = 0.0

K, dvec = get_coeffs_rhs(D, H, THETA, P1, P2, P3, P4, P5)

detK  = np.linalg.det(K)
condK = np.linalg.cond(K)
phi_3 = np.linalg.solve(K, dvec)

print("Determinant of K:", detK)
print("Condition number of K:", condK)
print("phi_3 = [", ", ".join(f"{x: .4f}" for x in phi_3), "]")

Determinant of K: 0.46566840006975907
Condition number of K: 6.542591657117212
phi_3 = [  0.5000,  0.0000, -1.1180, -0.0000,  0.5000,  0.0000, -1.1180, -0.0000,  0.5000, -1.1180 ]


### Task 4

Determine the member forces when $F$ is loaded by a unit force and all other joints are unloaded.

Define a Python variable `phi_4` to store $\boldsymbol{\phi}$ for this task.

In [6]:
D     = 1.0
H     = 6.0
THETA = 15.0

P1 = 0.0
P2 = 0.0
P3 = 0.0
P4 = 1.0
P5 = 0.0

K, dvec = get_coeffs_rhs(D, H, THETA, P1, P2, P3, P4, P5)

detK  = np.linalg.det(K)
condK = np.linalg.cond(K)
phi_4 = np.linalg.solve(K, dvec)
print("Determinant of K:", detK)
print("Condition number of K:", condK)
print("phi_4 = [", ", ".join(f"{x: .4f}" for x in phi_4), "]")

Determinant of K: 0.46566840006975907
Condition number of K: 6.542591657117212
phi_4 = [  0.7454, -0.3436, -0.2500,  0.5000,  0.5590, -0.5590, -0.0000,  1.0000, -0.0000, -0.0000 ]


### Task 5

Determine the member forces when $D$ is loaded by a unit force and all other joints are unloaded.

Define a Python variable `phi_5` to store $\boldsymbol{\phi}$ for this task.

In [7]:
D   = 1.0
H   = 6.0
THETA = 15.0

P1 = 0.0
P2 = 0.0
P3 = 0.0
P4 = 0.0
P5 = 1.0

K, dvec = get_coeffs_rhs(D, H, THETA, P1, P2, P3, P4, P5)

detK = np.linalg.det(K)
condK = np.linalg.cond(K)
phi_5 = np.linalg.solve(K, dvec)
print("Determinant of K:", detK)
print("Condition number of K:", condK)
print("phi_5 = [", ", ".join(f"{x: .4f}" for x in phi_5), "]")

Determinant of K: 0.46566840006975907
Condition number of K: 6.542591657117212
phi_5 = [  0.3727, -0.6872, -0.0000,  1.0000, -0.0000, -0.0000, -0.0000, -0.0000, -0.0000, -0.0000 ]


### Task 6

Determine the member forces
when $C$, $E$, $G$, $F$, and $D$
are loaded by unit forces.

Define a Python variable `phi_6` to store $\boldsymbol{\phi}$ for this task.

In [8]:
D   = 1.0
H   = 6.0
THETA = 15.0

P1 = 1.0
P2 = 1.0
P3 = 1.0
P4 = 1.0
P5 = 1.0

K, dvec = get_coeffs_rhs(D, H, THETA, P1, P2, P3, P4, P5)

detK = np.linalg.det(K)
condK = np.linalg.cond(K)
phi_6 = np.linalg.solve(K, dvec)
print("Determinant of K:", detK)
print("Condition number of K:", condK)
print("phi_6 = [", ", ".join(f"{x: .4f}" for x in phi_6), "]")

Determinant of K: 0.46566840006975907
Condition number of K: 6.542591657117212
phi_6 = [  0.1180, -3.3357, -1.9271,  2.6180,  0.3090, -1.8090, -1.1180,  1.0000,  0.5000, -1.1180 ]


### Task 7

Similar to Task 6,
determine the member forces
when $C$, $E$, $G$, $F$, and $D$
are loaded by unit forces,
but use the results in Tasks 1 - 5.

Define a Python variable `phi_7` to store $\boldsymbol{\phi}$ for this task.

In [9]:
D = 1.0
H = 6.0
THETA = 15.0

K1, d1 = get_coeffs_rhs(D, H, THETA, p1=1.0, p2=0.0, p3=0.0, p4=0.0, p5=0.0)
phi_1 = np.linalg.solve(K1, d1)

K2, d2 = get_coeffs_rhs(D, H, THETA, p1=0.0, p2=1.0, p3=0.0, p4=0.0, p5=0.0)
phi_2 = np.linalg.solve(K2, d2)

K3, d3 = get_coeffs_rhs(D, H, THETA, p1=0.0, p2=0.0, p3=1.0, p4=0.0, p5=0.0)
phi_3 = np.linalg.solve(K3, d3)

K4, d4 = get_coeffs_rhs(D, H, THETA, p1=0.0, p2=0.0, p3=0.0, p4=1.0, p5=0.0)
phi_4 = np.linalg.solve(K4, d4)

K5, d5 = get_coeffs_rhs(D, H, THETA, p1=0.0, p2=0.0, p3=0.0, p4=0.0, p5=1.0)
phi_5 = np.linalg.solve(K5, d5)

phi_7 = phi_1 + phi_2 + phi_3 + phi_4 + phi_5

K_all, d_all = get_coeffs_rhs(D, H, THETA, p1=1.0, p2=1.0, p3=1.0, p4=1.0, p5=1.0)
phi_6 = np.linalg.solve(K_all, d_all)

detK = np.linalg.det(K_all)
condK = np.linalg.cond(K_all)
print("Determinant of K:", detK)
print("Condition number of K:", condK)
print("phi_7 = [", ", ".join(f"{x: .4f}" for x in phi_7), "]")

Determinant of K: 0.46566840006975907
Condition number of K: 6.542591657117212
phi_7 = [  0.1180, -3.3357, -1.9271,  2.6180,  0.3090, -1.8090, -1.1180,  1.0000,  0.5000, -1.1180 ]


### Task 8

Determine the member forces using the assigned values of
$p_{1}$, $p_{2}$, $p_{3}$, $p_{4}$, and $p_{5}$.

Define a Python variable `phi_8` to store $\boldsymbol{\phi}$ for this task.

In [10]:
D   = 1.0
H   = 6.0
THETA = 15.0

P1 = 1.0
P2 = 9.0
P3 = 8.0
P4 = 7.0
P5 = 5.0 

K, dvec = get_coeffs_rhs(D, H, THETA, P1, P2, P3, P4, P5)

detK = np.linalg.det(K)
condK = np.linalg.cond(K)
phi_8 = np.linalg.solve(K, dvec)
print("Determinant of K:", detK)
print("Condition number of K:", condK)
print("phi_8 = [", ", ".join(f"{x: .4f}" for x in phi_8), "]")

Determinant of K: 0.46566840006975907
Condition number of K: 6.542591657117212
phi_8 = [  6.9142, -14.2923, -15.7254,  18.5623,  1.1631, -15.1631, -8.9443,  7.0000,  4.0000, -8.9443 ]


### Task 9

Similar to Task 8,
determine the member forces using the assigned values of
$p_{1}$, $p_{2}$, $p_{3}$, $p_{4}$, and $p_{5}$,
but use the results in Tasks 1 - 5.


Define a Python variable `phi_9` to store $\boldsymbol{\phi}$ for this task.

In [11]:
D = 1.0
H = 6.0
THETA = 15.0

P1 = 1.0
P2 = 9.0
P3 = 8.0
P4 = 7.0
P5 = 5.0

Kc, dc = get_coeffs_rhs(D, H, THETA, 1.0, 0.0, 0.0, 0.0, 0.0); phi_1 = np.linalg.solve(Kc, dc)
Ke, de = get_coeffs_rhs(D, H, THETA, 0.0, 1.0, 0.0, 0.0, 0.0); phi_2 = np.linalg.solve(Ke, de)
Kg, dg = get_coeffs_rhs(D, H, THETA, 0.0, 0.0, 1.0, 0.0, 0.0); phi_3 = np.linalg.solve(Kg, dg)
Kf, df = get_coeffs_rhs(D, H, THETA, 0.0, 0.0, 0.0, 1.0, 0.0); phi_4 = np.linalg.solve(Kf, df)
Kd, dd = get_coeffs_rhs(D, H, THETA, 0.0, 0.0, 0.0, 0.0, 1.0); phi_5 = np.linalg.solve(Kd, dd)

phi_9 = P1*phi_1 + P2*phi_2 + P3*phi_3 + P4*phi_4 + P5*phi_5

K_all, d_all = get_coeffs_rhs(D, H, THETA, P1, P2, P3, P4, P5)
phi_direct = np.linalg.solve(K_all, d_all)

print("Determinant of K:", detK)
print("Condition number of K:", condK)
print("phi_9 = [", ", ".join(f"{x: .4f}" for x in phi_9), "]")

Determinant of K: 0.46566840006975907
Condition number of K: 6.542591657117212
phi_9 = [  6.9142, -14.2923, -15.7254,  18.5623,  1.1631, -15.1631, -8.9443,  7.0000,  4.0000, -8.9443 ]


### Task 10

Determine the magnitudes of the support reactions
using the assigned values of
$p_{1}$, $p_{2}$, $p_{3}$, $p_{4}$, and $p_{5}$.

Define Python variables `rxn_A` and `rxn_B` 
to store the forces developed at $A$ and $B$, respectively.

In [12]:
D = 1.0
H = 6.0
THETA = 15.0

Ax_component = -phi_9[0]
rxn_A = abs(Ax_component)

b = atan((2 * H) / (3 * D))

Bx = phi_9[2] + phi_9[1] * cos(b)
By = phi_9[1] * sin(b)
rxn_B = sqrt(Bx**2 + By**2)

print(f"Magnitude of the reaction at A is: {rxn_A:.4f} kN")
print(f"Magnitude of the reaction at B is: {rxn_B:.4f} kN")

Magnitude of the reaction at A is: 6.9142 kN
Magnitude of the reaction at B is: 23.6766 kN


## Instructions

Do not use any library or module other than those in the Imports section.

### Parameters

For each set,
the values shown are that of
$d$, $h$, $\theta$, $p_{1}$, $p_{2}$, $p_{3}$, $p_{4}$, and $p_{5}$,
respectively.

- Set 1: 1.5 ft, 4.5 ft, 21°, 5 kN, 7 kN, 6 kN, 8 kN, 1 kN
- Set 2: 2.0 ft, 3.0 ft, 17°, 9 kN, 6 kN, 5 kN, 1 kN, 7 kN
- Set 3: 1.0 ft, 6.0 ft, 15°, 1 kN, 9 kN, 8 kN, 7 kN, 5 kN

### Scoring

In each of Tasks 1 - 9,
obtaining a correct member force merits one (1) point.

In each Task but 7, 9, and 10,
- obtaining a correct determinant merits one (1) point,
  and
- obtaining a correct condition number merits one (1) point.

For Task 10,
obtaining a correct reaction force merits five (5) points.

Every violation of an instruction
means a deduction of one (1) point.

All in all, one may earn up to 124 points for this activity.

### Submission

Download this notebook file,
and save with a filename following the pattern
`ACT-08_<Group name>`,
where the group name is as listed in class.
For example, if you belong to the group SixIsEven,
then your notebook should be named `ACT-08_SixIsEven.ipynb`.
Submit your notebook via the classwork platform for this activity in Google Classroom.
Submissions beyond the deadline will not be considered.

Lastly, the use of AI tools to answer this exam is not prohibited,
but it is of ethical interest to disclose such use.
This is in line with the
[MSU Policy on the Fair and Ethical Use of AI and Its Applications](https://www.msumain.edu.ph/wp-content/uploads/2024/05/MSU-Policy-on-Ethical-use-of-AI-Policies.pdf).
As such, please include a brief statement (in a private comment to this classwork)
declaring which and how AI tools are used in your work.

*Last updated by Christian Cahig on 2025-11-25*